### PR TITLE
Initial implementation of `dcgm` receiver version 2.

### PR DESCRIPTION
**Pull request by igorpeshansky**
## Description
Implements version 2 of the `dcgm` receiver with an initial set of DCGM GPU metrics as defined in [go/gce-gpu-dcgm-metrics-design](http://go/gce-gpu-dcgm-metrics-design).

## Related issue
[b/339105548](http://b/339105548)

## How has this been tested?
Integration tests, and manual run of the agent packages on VMs with NVIDIA H100 GPUs attached.

## Checklist:
- Unit tests
  - [ ] Unit tests do not apply.
  - [x] Unit tests have been added/modified and passed for this PR.
- Integration tests
  - [ ] Integration tests do not apply.
  - [x] Integration tests have been added/modified and passed for this PR.
- Documentation
  - [ ] This PR introduces no user visible changes.
  - [x] This PR introduces user visible changes and the corresponding documentation change has been made.
- Minor version bump
  - [ ] This PR introduces no new features.
  - [x] This PR introduces new features, and there is a separate PR to bump the [minor version](https://github.com/GoogleCloudPlatform/ops-agent/blob/master/VERSION) since the last [release](https://github.com/GoogleCloudPlatform/ops-agent/releases) already.
  - [ ] This PR bumps the version.

### CODE REVIEW

File: integration_test/third_party_apps_test/applications/dcgm/metadata.yaml [hunk 1]
```diff
@@ -106,9 +172,12 @@ configuration_options:
...
           description: A time.Duration value, such as 30s or 5m.
+        - name: receiver_version
+          default: "1"
+          description: Either 1 or 2. Version 2 has many more metrics available.
         - name: endpoint
           default: localhost:5555
           description: The DCGM service endpoint specified as 'hostname:port'
 minimum_supported_agent_version:
-  metrics: 2.38.0
+  metrics: 2.48.0
```
Comment (quentinmit) [inline integration_test/third_party_apps_test/applications/dcgm/metadata.yaml:182]: Why are you bumping this? The receiver existed before, too

File: integration_test/third_party_apps_test/applications/dcgm/metadata.yaml [hunk 2]
```diff
@@ -20,7 +20,7 @@ description: |-
   The NVIDIA DataCenter GPU Manager (DCGM) integration collects advanced GPU metrics,
```
Comment (quentinmit) [inline integration_test/third_party_apps_test/applications/dcgm/metadata.yaml:20]: Can we copy the NVML exercise script here to generate some meaningful metrics?

**Review by quentinmit — APPROVED** [review]
Approval modulo passing integration tests

File: apps/dcgm.go [hunk 3]
```diff
@@ -40,26 +41,166 @@ func (r MetricsReceiverDcgm) Pipelines(_ context.Context) ([]otel.ReceiverPipeli
 		r.Endpoint = defaultDcgmEndpoint
 	}
 
+	if r.ReceiverVersion == "2" {
+		return []otel.ReceiverPipeline{{
+			// TODO
```
Comment (LujieDuan) [inline apps/dcgm.go:46]:
```suggestion
```

File: integration_test/third_party_apps_test/applications/dcgmv1/metadata.yaml [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+    kind: GAUGE
+    description: Fraction of cycles data was being sent or received from GPU memory.
+    monitored_resources: [gce_instance]
+    labels:
+      - *model_label
+      - *uuid_label
+      - *gpu_number_label
+      - *instrumentation_source_label
+    notes:
+      - *dcgm_profiling_note
```
Comment (LujieDuan) [inline integration_test/third_party_apps_test/applications/dcgmv1/metadata.yaml:111]:
```suggestion
```
`dram_utilization` (bandwidth_utilization) now has fallback so we can remove the note.

File: integration_test/third_party_apps_test/applications/dcgm/metadata.yaml [hunk 5]
```diff
@@ -43,10 +43,10 @@ gpu_platforms: # p4, p100 don't emit DCGM profiling metrics
       - ubuntu-os-cloud:ubuntu-2004-lts
       - ubuntu-os-cloud:ubuntu-2204-lts
```
Comment (LujieDuan) [inline integration_test/third_party_apps_test/applications/dcgm/metadata.yaml:44]:
```suggestion
      - ubuntu-os-cloud:ubuntu-2204-lts
   - model: p100
    platforms:
      - ubuntu-os-cloud:ubuntu-2004-lts
   - model: p4
    platforms:
      - ubuntu-os-cloud:ubuntu-2004-lts
```
With the new receiver, P4 and P100 now can send DCGM metrics, so we should enable the DCGM test on those two GPUs.

**Review by LujieDuan — APPROVED** [review]
LGTM! Couple of small suggestions:

File: integration_test/third_party_apps_test/applications/dcgmv1/metadata.yaml (hunk 4 above)
Comment (igorpeshansky) [reply integration_test/third_party_apps_test/applications/dcgmv1/metadata.yaml]: So it does. I forgot that v1 and v2 reuse the same underlying implementation. Thanks for catching this.

File: integration_test/third_party_apps_test/applications/dcgmv1/metadata.yaml [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+      - ubuntu-os-cloud:ubuntu-2004-lts
+  - model: l4
+    platforms:
+      - debian-cloud:debian-11
+      - ml-images:common-gpu-debian-11-py310
+      - rocky-linux-cloud:rocky-linux-8
+      - rocky-linux-cloud:rocky-linux-9
+      - suse-cloud:sles-15
+      - ubuntu-os-cloud:ubuntu-2004-lts
+      - ubuntu-os-cloud:ubuntu-2204-lts
```
Comment (igorpeshansky) [inline integration_test/third_party_apps_test/applications/dcgmv1/metadata.yaml:44]: @LujieDuan Replicating [your comment](https://github.com/GoogleCloudPlatform/ops-agent/pull/1758/files/98ac11fd7f47de4ed6073d64f90c105a75f1c9e7#r1763717461) here.
```suggestion
      - ubuntu-os-cloud:ubuntu-2204-lts
  - model: p100
    platforms:
      - ubuntu-os-cloud:ubuntu-2004-lts
  - model: p4
    platforms:
      - ubuntu-os-cloud:ubuntu-2004-lts
```
With the new receiver, P4 and P100 now can send DCGM metrics, so we should enable the DCGM test on those two GPUs.

File: integration_test/third_party_apps_test/applications/dcgm/metadata.yaml (hunk 5 above)
Comment (igorpeshansky) [reply integration_test/third_party_apps_test/applications/dcgm/metadata.yaml:44]: Unfortunately, most profiling metrics are actually not available on P4 and P100, so I've had to revert this change.

File: integration_test/third_party_apps_test/applications/dcgmv1/metadata.yaml (hunk 6 above)
Comment (igorpeshansky) [reply integration_test/third_party_apps_test/applications/dcgmv1/metadata.yaml:44]: Unfortunately, most profiling metrics are actually not available on P4 and P100, so I've had to revert this change.

**Pull request merged**